# HG changeset patch
# User Rik <rik@octave.org>
# Date 1475018664 25200
#      Tue Sep 27 16:24:24 2016 -0700
# Node ID b9f5ac691e03ebdafc9b34d7002521100edcea4e
# Parent  e0a1e8803d8c29dd2824dc56c3603be2c28133c7
colormap.m: Emit clearer error when map name does not exist (bug #49202).

* colormap.m: Use try/catch block to 'feval (map)'.  If map does not exist then
issue a clear error mesage in catch block.  Add %!error test to validate new
behavior.

diff --git a/scripts/image/colormap.m b/scripts/image/colormap.m
--- a/scripts/image/colormap.m
+++ b/scripts/image/colormap.m
@@ -67,17 +67,21 @@ function cmap = colormap (varargin)
   endif
 
   if (nargin == 1)
     map = varargin{1};
     if (ischar (map))
       if (strcmp (map, "default"))
         map = viridis (64);
       else
-        map = feval (map);
+        try
+          map = feval (map);
+        catch
+          error ("colormap: failed to set MAP <%s>", map);
+        end_try_catch
       endif
     endif
 
     if (! isempty (map))
       if (! (isnumeric (map) && isreal (map)
              && ndims (map) == 2 && columns (map) == 3))
         error ("colormap: MAP must be a real-valued N x 3 ([R,G,B]) matrix");
       endif
@@ -137,10 +141,10 @@ endfunction
 ## Test input validation
 %!error colormap (1,2,3)
 %!error <MAP must be a real-valued N x 3> colormap ({1,2,3})
 %!error <MAP must be a real-valued N x 3> colormap ([1 i 1])
 %!error <MAP must be a real-valued N x 3> colormap (ones(3,3,3))
 %!error <MAP must be a real-valued N x 3> colormap ([1 0 1 0])
 %!error <all MAP values must be in the range> colormap ([-1 0 0])
 %!error <all MAP values must be in the range> colormap ([2 0 0])
-%!error colormap ("invalid", "name")
+%!error <failed to set MAP .invalid_map_name.> colormap ("invalid_map_name")
 
